python/createCollectionDownload.py

```diff
@@ -2,75 +2,189 @@
+    r = request_api(url=url, headers=headers)
+    if not isinstance(collection_type, str) or collection_type.lower() not in ['chain', 'center']:
+        raise ValueError(f"Collection type '{collection_type}' must be one of ['chain', 'center'].")
+    
+    if cache_time < 0:
+        raise ValueError("Please enter a valid duration for cache_time that is an int > 0.")
+
+def get_collection_urls(collection_id:str, collection_type:str):
+    """
+    Returns URLs and url params for requesting the ChainXY API for collection records based on collection_type.
+    """
+    records_url = check_url = new_download_url = ''
+
+    if collection_type == 'chain':
+        
+        records_url = f'https://location.chainxy.com/api/Downloads?fields=Id,ChainList,Label,User,ReportType,CreateDate,DataDate,Status,Format,Count,Link,DisplayValue,&Query=%7B%22ChainListId%22:{collection_id}%7D&OrderBy=-CreateDate'
+        check_url = f'https://location.chainxy.com/api/ChainLists/{collection_id}'
+        new_download_url = f"https://location.chainxy.com/api/ChainLists/Download/{collection_id}"
 
+        url_params = {
+            "format": "CSV",  # ZIP_CSV also works
+            "splitLayers": "false",
+            # "dataDate": "2019-10-03" # OPTIONAL
+        }
 
-def download_collection(cxy_api_key:str, collection_id:int):
+    elif collection_type == 'center':
+        
+        records_url = f'https://location.chainxy.com/api/Downloads?fields=Id,SiteList,Label,User,ReportType,CreateDate,DataDate,Status,Format,Count,Link,DisplayValue,&Query=%7B%22SiteListId%22:{collection_id}%7D&OrderBy=-CreateDate'
+        check_url = f'https://location.chainxy.com/api/SiteLists/{collection_id}'
+        new_download_url =  f'https://location.chainxy.com/api/SiteLists/Download/{collection_id}'
+
+        url_params = {
+            "format": "ZIP_CSV"
+        }
+    
+    return records_url, check_url, new_download_url, url_params
+
+def retrieve_new_download(create_date, cache_time:int):
     """
-    Downloads a chainxy collection based on the provided collection ID.
-    Params:
-    cxy_api_key:str - ChainXY API Key
-    collection_id:str - ID of the collection for which a new download will be initiated.
+    Returns true if a new download should be created. 
     """
+    dt = datetime.strptime(create_date, '%Y-%m-%dT%H:%M:%S.%f')
+    now = datetime.utcnow()
+    td = (abs(dt - now)).total_seconds()/3600
+    return td > cache_time
     
-    check_api_key(cxy_api_key)
+def check_record_exists(url:str, headers:str):
+    """
+    Returns true if a record of the collection exists in the database.
+    """
+    r = request_api(url=url, headers=headers)
+    result = json.loads(r.text)
+    if 'Message' in result:
+        if 'record does not exist' in result['Message'].lower():
+            print("No collection found. Please ensure you have the correct collection ID and collection Type.")
```

Comment:
this should raise a ValueError in this context (i.e. error out if collection doesn't exist)

Reply:
Changed to ValueError to make the program simply error out if the collection doesn't exist